lib/delayed/worker.rb

```diff
@@ -101,12 +101,15 @@ def work_off(num = 100)
         pool = Concurrent::FixedThreadPool.new(jobs.length)
         jobs.each do |job|
           pool.post do
-            self.class.lifecycle.run_callbacks(:thread, self, job) do
+            self.class.lifecycle.run_callbacks(:thread, self) do
               success.increment if perform(job)
+            rescue DeserializationError => e
```

Comment:
These two `rescue`s exist as a catch-all, just in case job cleanup is still possible at this point, but I was unable to actually exercise these codepaths because the ones in `perform` happen at the very end of that method now.

Reply:
By clean up, I mean update run_at/attempts/failed_at/etc -- all the clean up work necessary to enable exponential backoff and prevent zombie jobs from building up and exhausting worker resources.

Reply:
For posterity, i think we can punt on this change for now to get this fix in, and then revisit!